Source/JavaScriptCore/wasm/WasmModuleInformation.h

```diff
@@ -99,6 +99,10 @@ struct ModuleInformation : public ThreadSafeRefCounted<ModuleInformation> {
     bool isDeclaredException(uint32_t index) const { return m_declaredExceptions.contains(index); }
     void addDeclaredException(uint32_t index) { m_declaredExceptions.set(index); }
 
+    bool isSIMDFunction(uint32_t index) const { Locker locker { m_simdLock }; ASSERT(index <= internalFunctionCount()); ASSERT(m_seenFunctions.contains(index)); return m_simdFunctions.contains(index); }
+    void addSIMDFunction(uint32_t index) { Locker locker { m_simdLock }; ASSERT(index <= internalFunctionCount()); ASSERT(!m_seenFunctions.contains(index)); m_simdFunctions.set(index); }
+    void doneSeeingFunction(uint32_t index) { Locker locker { m_simdLock }; m_seenFunctions.set(index); }
+
```

Comment:
See comment. And why we needed a lock is that BitVector is updated per-word-unit. So if multiple concurrent thread updates the same word, since update is LOAD-OR-STORE sequence, concurrently doing that will miss some bits. WTF::Bitmap has concurrent set via `testAndSet`, but BitVector does not have that.
But we do not need to use a lock and BitVector. Let's just have this field in FunctionData.